public/src/pages/TeacherPage.tsx

```diff
@@ -383,4 +378,21 @@ export class TeacherPage extends ViewPage {
         return approve ? "approve" : "undo approval for";
     }
 
+    private async fetchCourseRepos(courseID: number): Promise<Map<Repository.Type, string>> {
+        return this.courseMan.getRepositories(courseID,
+                [Repository.Type.COURSEINFO,
+                Repository.Type.ASSIGNMENTS,
+                Repository.Type.TESTS,
+                Repository.Type.SOLUTIONS]);
+    }
+
+    private setupRepos(): Map<number, Map<Repository.Type, string>> {
+        const courses = this.courses;
```

Comment:
Nice func, but do you need to copy `this.courses` here?

Reply:
Not really.